### DISCUSSION TITLE
nvim-dbee vs dadbod

### DISCUSSION THREAD

**Discussion by Demianeen** [kndndrj/nvim-dbee#119] [Q&A] [answered]
I recently started to learn sql, so I started to look for options to execute some queries and view the database right inside nvim. I found two options that I couldn't find any comparison on. This plugin and dadbod.

Can somebody explain when to use which one? Or it doesn't really matter?

**Comment by andradei** [comment, marked as answer]
It doesn't really matter. Here is a vague comparison:

- They largely implement the same features with minor differences (connect to DB's, write and save queries, display the result in a table view, autocompletion via external plugins)
- I'm assuming that `nvim-dbee` might be much faster due to it's core being written in Go vs `dadbod`'s vimscript (don't take my work for it, also performance at the plugin level might not be that important)
- I find `nvim-dbee`'s UI a tiny bit nicer than https://github.com/kristijanhusak/vim-dadbod-ui
- `nvim-dbee` comes with UI in the plugin while dadbod requires an extra plugin (mentioned above) but that's not a big deal
- I don't know about `dadbod`, but `nvim-dbee` has an export to CSV/JSON feature that's very useful
- `dadbod` has been around for much longer, was written by one of the best vim plugin creators, and is very mature and battle-tested

**Reply by andradei** [reply]
If this plugin keep being actively developed, it could become the neovim successor of dadbod. I'm favoring it for now. But you can't go wrong with dadbod, that plugin is very stable!

**Comment by ArkadyBuryakov** [comment]
@andradei gave pretty good review.

I would add that nvim-dbee and dadbod ecosystem just feel different.

DBee feels like encapsulated experience. It opens in a new tab and allows you to work within it's own global project.
Really cool features:
- Export options (JSON + CSV for selected lines or whole result)
- Change database within connection
- Result pagination

Dadbod is a plugin that allows you to execute your buffer or selected lines and show result in a bottom panel.
Dadbod UI is just a side panel that allows you to configure your database connections and navigate between them.
It lacks some functionality like export, however it is more flexible on UI organization and could be integrated with your current projects.
